scintillometry/io/psrfits/hdu.py

```diff
@@ -93,22 +143,53 @@ def frequency(self):
         # and c_freq is the frequency of channel n_nchan / 2.  We use
         # (n_chan + 1) // 2 to ensure this makes sense for n_chan = 1
         # and is consistent with the document at least for even n_chan.
+
         freq = c_freq + (np.arange(1, n_chan + 1) -
                          ((n_chan + 1) // 2)) * chan_bw
         return u.Quantity(freq, u.MHz, copy=False)
 
+    @frequency.setter
+    def frequency(self, freq):
+        """Frequency setter. The input should be the frequency array."""
+        freq = freq.to_value(u.MHz)
+        n_chan = len(freq)
+        # add the channel 0
+        # we assuem the frequency resoluation is the same across the band.
+        freq_pad = np.insert(freq, 0, 2 * freq[0] - freq[1])
+        # Are we doing the right thing? Maybe we should check with psrfits.
+        c_chan = freq_pad[((n_chan + 1) // 2)]
+        bw = freq_pad.ptp()
+        self.hdu.header['OBSNCHAN'] = n_chan
+        self.hdu.header['OBSFREQ'] = c_chan
+        self.hdu.header['OBSBW'] = bw
+
     @property
     def ra(self):
         return Longitude(self.header['RA'], unit=u.hourangle)
 
+    @ra.setter
+    def ra(self, val):
+        self.hdu.header['RA'] = val.to_string(sep=':', pad=True)
+
     @property
     def dec(self):
         return Latitude(self.header['DEC'], unit=u.deg)
 
+    @dec.setter
+    def dec(self, val):
+        self.hdu.header['DEC'] = val.to_string(sep=':', alwayssign=True,
+                                               pad=True)
+
     @property
     def obs_mode(self):
         return self.header['OBS_MODE']
 
+    @obs_mode.setter
+    def obs_mode(self, val):
+        assert val in ['PSR', 'CAL', 'SEARCH'], \
```

Comment:
Total nitpick here, but a `set` is faster generally, i.e., do `{'PSR', 'CAL', 'SEARCH'}`